decidim-proposals/lib/decidim/content_renderers/proposal_renderer.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ContentRenderers
+    # A renderer that searches Global IDs representing proposals in content
+    # and replaces it with a link to their show page.
+    #
+    # e.g. gid://<APP_NAME>/Decidim::Proposals::Proposal/1
+    #
+    # @see BaseRenderer Examples of how to use a content renderer
+    class ProposalRenderer < BaseRenderer
+      # Matches a global id representing a Decidim::User
+      GLOBAL_ID_REGEX = %r{gid:\/\/([\w-]*\/Decidim::Proposals::Proposal\/(\d+))}i
+
+      # Replaces found Global IDs matching an existing proposal with
+      # a link to its show page. The Global IDs representing an
+      # invalid Decidim::Proposals::Proposal are replaced with '???' string.
+      #
+      # @return [String] the content ready to display (contains HTML)
+      def render
+        content.gsub(GLOBAL_ID_REGEX) do |proposal_gid|
+          begin
+            proposal = GlobalID::Locator.locate(proposal_gid)
+            Decidim::Proposals::ProposalPresenter.new(proposal).display_mention
+          rescue ActiveRecord::RecordNotFound
+            "<???>"
```

Comment:
We should display a meaningful text when a proposal couldn't be found (or just the id), but ??? isn't really helpful

Reply:
Should i replace <???> for <1234>? or you prefer <the proposal that was referenced here doesn't exist anymore>?

Reply:
Maybe @decidim/product should decide about this? (but the second option sounds better)

Reply:
I've finally opted for "the proposal that was referenced here with id %{proposal_id} but doesn't exist anymore".
Done!